lib/Access.php

```diff
@@ -530,38 +529,36 @@ public function groupsMatchFilter($groupDNs) {
 	 * returns the internal ownCloud name for the given LDAP DN of the user, false on DN outside of search DN or failure
 	 *
 	 * @param string $fdn the dn of the user object
-	 * @param string $ldapName optional, the display name of the object
 	 * @return string|false with with the name to use in ownCloud
 	 * @throws \OC\ServerNotAvailableException
 	 */
-	public function dn2username($fdn, $ldapName = null) {
+	public function dn2username($fdn) {
 		//To avoid bypassing the base DN settings under certain circumstances
 		//with the group support, check whether the provided DN matches one of
 		//the given Bases
 		if (!$this->isDNPartOfBase($fdn, $this->connection->ldapBaseUsers)) {
 			return false;
 		}
 
-		return $this->dn2ocname($fdn, $ldapName, true);
+		return $this->dn2ocname($fdn, true);
 	}
 
 	/**
 	 * returns an internal ownCloud name for the given LDAP DN, false on DN outside of search DN
 	 *
 	 * @param string $fdn the dn of the user object
-	 * @param string $ldapDisplayName optional, the display name of the object
 	 * @param bool $isUser optional, whether it is a user object (otherwise group assumed)
 	 * @return string|false with with the name to use in ownCloud
 	 * @throws \BadMethodCallException
 	 * @throws \OC\ServerNotAvailableException
 	 */
-	public function dn2ocname($fdn, $ldapDisplayName = null, $isUser = true) {
+	public function dn2ocname($fdn, $isUser = true) {
 		if ($isUser) {
 			$mapper = $this->getUserMapper();
-			$displayNameAttribute = $this->connection->ldapUserDisplayName;
+			$nameAttribute = (string)$this->connection->ldapExpertUsernameAttr;
 		} else {
 			$mapper = $this->getGroupMapper();
-			$displayNameAttribute = $this->connection->ldapGroupDisplayName;
+			$nameAttribute = (string)$this->connection->ldapExpertGroupnameAttr;
 		}
```

Comment:
add comment why not using ldapUserDisplayName/ldapGroupDisplayName

Reply:
I don't think there is any need. The internal ownCloud name should be internal, that's what the username / groupname attribute is for. A displayname is meant to be displayed, so using it for something internal doesn't make too much sense.

For the code changes, the user displayname was requested but not used (the username attribute was used instead), and the group displayname doesn't make sense as internal name as said.

Reply:
but this does not brake anything right ?

Reply:
I don't think so. There is a migration in place to copy the old displayname attribute (which usually is the "cn") into the new groupname attribute, so there won't be any noticeable change after the upgrade.

For new installation, the groupname will default to the uuid, which should be the entryuuid or entryguid (maybe others, depending on the provider), but it can be adjusted if needed in the wizard.